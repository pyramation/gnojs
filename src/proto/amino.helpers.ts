import Long from "long";
export interface AminoHeight {
  readonly revision_number?: string;
  readonly revision_height?: string;
}
export function omitDefault<T extends string | number | Long>(input: T): T | undefined {
  if (typeof input === "string") {
    return input === "" ? undefined : input;
  }

  if (typeof input === "number") {
    return input === 0 ? undefined : input;
  }

  if (Long.isLong(input)) {
    return input.isZero() ? undefined : input;
  }

  throw new Error(`Got unsupported type ${typeof input}`);
}