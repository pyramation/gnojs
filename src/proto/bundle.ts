import * as _0 from "./github.com/gnolang/gno/pkgs/bft/abci/types/abci";
import * as _1 from "./github.com/gnolang/gno/pkgs/crypto/ed25519/ed25519";
import * as _2 from "./github.com/gnolang/gno/pkgs/crypto/merkle/merkle";
import * as _3 from "./github.com/gnolang/gno/pkgs/crypto/multisig/multisig";
import * as _4 from "./github.com/gnolang/gno/pkgs/crypto/secp256k1/secp256k1";
import * as _5 from "./google/protobuf/any";
import * as _6 from "./google/protobuf/timestamp";
export const abci = { ..._0
};
export const tm = { ..._1,
  ..._2,
  ..._3,
  ..._4
};
export namespace google {
  export const protobuf = { ..._5,
    ..._6
  };
}