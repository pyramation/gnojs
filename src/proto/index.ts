export * from "./gno/bundle";
export * from "./abci/bundle";
export * from "./tm/bundle";
export * from "./bank/bundle";
export * from "./vm/bundle";
export * from "./std/bundle";
export * from "./google/bundle";