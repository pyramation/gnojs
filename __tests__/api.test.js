import { abci } from '../src/proto';
import { tm } from '../src/proto';

it('abci', async () => {
  const result = abci.BlockParams.toJSON({
    MaxBlockBytes: '',
    MaxGas: '',
    MaxDataBytes: '',
    MaxTxBytes: '',
    TimeIotaMS: '000'
  });
  expect(result).toMatchSnapshot();
});

it('tm', async () => {
  const result = tm.Proof.toJSON({
    Ops: [
      tm.ProofOp.toJSON({
        Data: undefined,
        Key: undefined,
        Type: ''
      })
    ]
  });
  expect(result).toMatchSnapshot();
});
