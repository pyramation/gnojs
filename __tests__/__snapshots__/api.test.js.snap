// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`abci 1`] = `
Object {
  "MaxBlockBytes": "0",
  "MaxDataBytes": "0",
  "MaxGas": "0",
  "MaxTxBytes": "0",
  "TimeIotaMS": "000",
}
`;

exports[`tm 1`] = `
Object {
  "Ops": Array [
    Object {
      "Type": "",
    },
  ],
}
`;
